src/ipc/handlers/app_handlers.ts

```diff
@@ -83,40 +83,57 @@ async function executeApp({
   appId,
   event, // Keep event for local-node case
   isNeon,
+  installCommand,
+  startCommand,
 }: {
   appPath: string;
   appId: number;
   event: Electron.IpcMainInvokeEvent;
   isNeon: boolean;
+  installCommand?: string | null;
+  startCommand?: string | null;
 }): Promise<void> {
   if (proxyWorker) {
     proxyWorker.terminate();
     proxyWorker = null;
   }
-  await executeAppLocalNode({ appPath, appId, event, isNeon });
+  await executeAppLocalNode({
+    appPath,
+    appId,
+    event,
+    isNeon,
+    installCommand,
+    startCommand,
+  });
 }
 
 async function executeAppLocalNode({
   appPath,
   appId,
   event,
   isNeon,
+  installCommand,
+  startCommand,
 }: {
   appPath: string;
   appId: number;
   event: Electron.IpcMainInvokeEvent;
   isNeon: boolean;
+  installCommand?: string | null;
+  startCommand?: string | null;
 }): Promise<void> {
-  const spawnedProcess = spawn(
-    "(pnpm install && pnpm run dev --port 32100) || (npm install --legacy-peer-deps && npm run dev -- --port 32100)",
-    [],
-    {
-      cwd: appPath,
-      shell: true,
-      stdio: "pipe", // Ensure stdio is piped so we can capture output/errors and detect close
-      detached: false, // Ensure child process is attached to the main process lifecycle unless explicitly backgrounded
-    },
-  );
+  const defaultCommand =
+    "(pnpm install && pnpm run dev --port 32100) || (npm install --legacy-peer-deps && npm run dev -- --port 32100)";
+  const command =
```

Comment:
This feels convoluted to me, I think both *install* and *start* should be defined or they should both be undefined. I think pretty much every standard web app requires both.